client/my-sites/sidebar/sidebar.jsx

```diff
@@ -660,12 +661,22 @@ export class MySitesSidebar extends Component {
 			return null;
 		}
 
+		const adminUrl =
+			this.props.isJetpack && ! this.props.isAtomicSite && ! this.props.isVip
+				? formatUrl(
```

Comment:
Why do we need `formatUrl` here (but not e.g. on the other branch of the ternary)?

Reply:
We parse, set the query and hash, then format in this branch. In the other branch, we just use the admin_url as provided (which was the behavior before). We need `format` to get back to a string URL from the parsed pieces. This is essentially `add_query_args` with a hash as well.